### ISSUE TITLE
block-opening-brace-space-before off-by-one in CRLF file

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#1502]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`block-opening-brace-space-before` is reporting the wrong column on a CRLF file.

> Which rule, if any, is this issue related to?

`block-opening-brace-space-before`

> What CSS is needed to reproduce this issue?

```css
@keyframes animation-show
{
from {transform: translateY(100%);}
to {transform: translateY(0);}
}
```

A folder that reproduces this can be found [here](https://github.com/stylelint/stylelint/files/324167/linter-stylelint_GH___.zip).

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "block-opening-brace-space-before": "always"
  }
}
```

> Which version of stylelint are you using?

`6.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Through the CLI and Node API (Atom v.1.9.0-beta0).

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

The correct column reported regardless of the EOL style of the file.

> What actually happened (e.g. what warnings or errors you are getting)?

A column past the end of the line is reported when the file is a `CRLF` format.

_Note: Originally filed on the wrong repo by @dawnwish [here](https://github.com/AtomLinter/linter-eslint/issues/616)._

**Comment by ntwb** [comment]
I was going to add some tests but we already have one for this [here](https://github.com/stylelint/stylelint/blob/master/src/rules/block-opening-brace-space-before/__tests__/index.js#L44-L48)

```js
    code: "a\r\n{ color: pink; }",
    description: "CRLF",
    message: messages.expectedBefore(),
    line: 1,
    column: 3,
```

Whereas the report above is expecting the warning to be `line: 2, column: 1,`

Similarly the test [here](https://github.com/stylelint/stylelint/blob/master/src/rules/block-opening-brace-space-before/__tests__/index.js#L39-L42) for a newline `\n` without `CRLF` behaves similar: 
```js
    code: "a\n{ color: pink; }",
    message: messages.expectedBefore(),
    line: 1,
    column: 2,
```

The expected warning above should also be `line: 2, column: 1,`

To make things a little weirder, the following CSS:
```css
a
 { color: pink; }
```

That has a space before the opening brace and reports `2:1   ✖  Expected single space before "{"   block-opening-brace-space-before` which is the correct location, but incorrect because there is a space before the opening brace.

**Comment by jeddy3** [comment]
@ntwb Thanks for looking into this.

I think this test is wrong:

```js
    code: "a\r\n{ color: pink; }",
    description: "CRLF",
    message: messages.expectedBefore(),
    line: 1,
    column: 3,
```

It should be:

```js
    code: "a\r\n{ color: pink; }",
    description: "CRLF",
    message: messages.expectedBefore(),
    line: 1,
    column: 2,
```

To resolve this issue I think we need to detect the use of CRLF and make an adjustment to the column number.

> To make things a little weirder, the following CSS:

This is the intended behaviour. It is expecting a *single* space between the end of the selector and the opening brace, but it found a *newline and* a space.

**Comment by ntwb** [comment]
But shouldn't the warning be line `2`, and not line `1` as the brace `{` is on line `2`?

**Comment by Arcanemagus** [comment]
Shouldn't the error be marked as where the problem is (the extra space/newline chars) though? Putting it on the `{` doesn't really make sense, to me at least, since the problem isn't with the `{`, but the space (or lack thereof) _before_ it (in this rule, the same logic applies elsewhere).

**Comment by jeddy3** [comment]
> Putting it on the { doesn't really make sense, to me at least, since the problem isn't with the {, but the space (or lack thereof) before it (in this rule, the same logic applies elsewhere).

This is correct.
